sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala

```diff
@@ -431,9 +431,11 @@ class DataFrameReaderWriterSuite extends QueryTest with SharedSQLContext with Be
 
     // when users specify the schema
     val inputSchema = new StructType().add("s", IntegerType, nullable = false)
+    val expectedSchema = StructType(StructField("i", IntegerType, nullable = false) :: Nil)
```

Comment:
Ideally we should print out the problematic field then during comparison.
Spark has been using `catalogString` for now for simplicity. Let's stick to `catalogString` for now to be consistent.